### PR TITLE
Several improvements based on OVN work

### PR DESCRIPTION
**Pull request by ryzhyk**
Several improvements based on OVN work:

- allow DDlog libraries to add new Rust dependencies
- `min` and `max` stdlib functions
- improved allocator API
- `deref()` function
- fix the antijoin implementation

### CODE REVIEW

File: lib/allocate.dl [hunk 1]
```diff
@@ -1,19 +1,21 @@
-// Allocate a unique 64-bit identifier in the range `1..max_val` for each element in `toallocate`.
+// Allocate a unique identifier in the range `min_val..max_val` for each element in `toallocate`.
```
Comment (mbudiu-vmw) [inline lib/allocate.dl]: there is some constraint on the type 'N. What is it?

File: lib/allocate.rs [hunk 2]
```diff
@@ -1,39 +1,78 @@
...
     for b in toallocate.x.iter() {
         loop {
-            last = if last + 1 <= *max_val {last + 1} else {1};
-            if last == first {
-                return res;
-            };
-            if allocated.x.contains(&last) { continue; };
-            res.x.push(((*b).clone(), last));
-            break;
+            next = if next == *max_val { *min_val } else { next + N::one() };
```
Comment (mbudiu-vmw) [inline lib/allocate.rs]: won't this get into an infinite loop if all values are allocated?

File: lib/allocate.dl (hunk 1 above)
Comment (ryzhyk) [reply lib/allocate.dl]: ack

File: lib/allocate.rs (hunk 2 above)
Comment (ryzhyk) [reply lib/allocate.rs]: It uses `offset` to terminate the loop, so should be fine.

**Pull request merged**